shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-distsql/shardingsphere-sharding-distsql-handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/checker/ShardingTableRuleStatementChecker.java

```diff
@@ -162,12 +162,12 @@ private static Collection<String> parseDateSource(final String dateSource) {
     }
     
     private static Collection<String> getDataSourceNames(final Collection<String> actualDataNodes) {
-        return actualDataNodes.stream().map(each -> {
-            if (isValidDataNode(each)) {
-                return actualDataNodes.stream().map(each1 -> new DataNode(each1).getDataSourceName()).collect(Collectors.toList());
-            }
-            return Collections.singletonList(each);
-        }).flatMap(Collection::stream).collect(Collectors.toCollection(LinkedHashSet::new));
+        Collection<String> result = new HashSet<>();
+        for (String each : actualDataNodes) {
+            String dataSourceName = isValidDataNode(each) ? new DataNode(each).getDataSourceName() : each;
+            result.add(dataSourceName);
```

Comment:
It is better to inline dataSourceName and add()

Reply:
> It is better to inline dataSourceName and add()

Thank you, I will optimize it.